src/Microsoft.ML.HalLearners/OlsLinearRegression.cs

```diff
@@ -35,7 +35,6 @@
 namespace Microsoft.ML.Trainers.HalLearners
 {
     /// <include file='doc.xml' path='doc/members/member[@name="OLS"]/*' />
-    [BestFriend]
```

Comment:
>BestFriend [](start = 5, length = 10)

Out of curiosity, can we modify analyzer to complain about usage of [BestFriend] on top of public class?
Unrelated to PR, just wondering.

Reply:
We can indeed. Though as you suggest, extremely low priority. Entered anyway as #2415.